adapters/faraday/lib/opentelemetry/adapters/faraday.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module Faraday
+      module_function
+
+      TRACER_NAME = 'faraday'
+      TRACER_VERSION = Gem.loaded_specs[TRACER_NAME].version.to_s
+
+      def install(config = {name: TRACER_NAME, version: TRACER_VERSION})
```

Comment:
If at all possible, I'd like to see the auto-installer take care of binding the tracer to instrumentation. I think the install method should take a reference to the tracer and a config. We can talk through this more to see if there are any complications with this scheme.

Reply:
I've made changes to allow for `config[:tracer]`, which overrides the `default_tracer` implementation (in `api/lib/opentelemetry/adapter`).